File: kotlin/lesson 4/CountingSort.kt
fun countingSort(nums: IntArray, max: Int) {
    val freq = IntArray(max + 1)
    for (num in nums) {
        freq[num]++
    }

    var current = 0
    for (i in 0..freq.size-1) {
        for (count in 0..freq[i]-1) {
            nums[current] = i
            current++
        }
    }
}

File: python/lesson 4/counting_sort.py
from typing import List


class Solution:
    def countingSort(self, nums: List[int], max: int):
        freq = [0] * (max + 1)
        for num in nums: freq[num] += 1
        
        current = 0    
        for i, f in enumerate(freq):
            for _ in range(f):
                nums[current] = i
                current += 1
